threedworld-mit/tdw
# Added set_ui_element_position and a mask example controller

## How to test

1. Clone this repo
2. Install TDW locally: `cd tdw/Python ; git checkout set_ui_element_position ; pip3 install -e .`
3. Download and extract the test build: https://www.dropbox.com/scl/fi/bzkkujcoemutc9nxey46p/TDW_Windows_set_ui_element_position.zip?rlkey=90awlbokp6vsuwkj1ne61wt42&dl=0
4. The test controller is located in: `tdw/Python/example_controllers/ui/mask.py`
5. Start the controller
6. Manually start the test build

The example controller will show you how to apply a "mask" to a third-person camera. I didn't write an example of how to do this with a VR camera, but the process should be nearly identical.

## v1.12.25

### Command API

#### New Commands

| Command | Description |
| --- | --- |
| `set_ui_element_position` | Set the position of a UI element. |

### `tdw` module

- Added: `ui.set_position(id, position)`.

### Example Controllers

- Added: `ui/mask.py`

### Documentation

#### Modified Documentation

| Document | Modification |
| --- | --- |
| `lessons/ui/ui.md` | Added a section describing how to create an image "mask".<br>Added a section describing how to move an image. |
